File: banchus-packet/src/main/java/pe/nanamochi/banchus/packets/client/ChannelLeavePacket.java
package pe.nanamochi.banchus.packets.client;

import java.io.IOException;
import java.io.InputStream;
import lombok.Data;
import lombok.NoArgsConstructor;
import pe.nanamochi.banchus.core.ClientPacket;
import pe.nanamochi.banchus.core.Packets;
import pe.nanamochi.banchus.io.IDataReader;

@Data
@NoArgsConstructor
public class ChannelLeavePacket implements ClientPacket {
  private String name;

  @Override
  public Packets getPacketType() {
    return Packets.OSU_CHANNEL_LEAVE;
  }

  @Override
  public void read(IDataReader reader, InputStream stream) throws IOException {
    this.name = reader.readString(stream);
  }
}

File: banchus-packet/src/main/kotlin/pe/nanamochi/banchus/packets/client/ChannelLeavePacket.kt
package pe.nanamochi.banchus.packets.client

import java.io.InputStream
import pe.nanamochi.banchus.core.ClientPacket
import pe.nanamochi.banchus.core.PacketType
import pe.nanamochi.banchus.io.DataReader

class ChannelLeavePacket : ClientPacket() {
    override val type = PacketType.OSU_CHANNEL_LEAVE

    var name: String = ""

    override fun read(reader: DataReader, ins: InputStream) {
        this.name = reader.readString(ins)
    }
}
